

<launch>
  <include file="$(find turtlebot_bringup)/launch/minimal.launch" />

  <include file="$(find adaptive_local_planner)/launch/includes/move_base.launch.xml" />


   <node pkg="pose_publisher" type="pose_publisher" name="pose_publisher" output="screen">
        <param name="map_frame" value="/map" />
         <param name="base_frame" value="/base_link" />
        <param name="pose_topic_name" value="/posegmapping" />
        <param name="publish_frequency" value="10.0" />

  </node>

    <node  pkg="urg_node" type="urg_node" name="urg_node" >
     <param name="angle_min" value="-1.6"/>
     <param name="angle_max" value="1.6"/>
  </node>
  <node pkg="tf" type="static_transform_publisher" name="link1_broadcaster" args="0.110 0 0.02478 0 0 0 1 base_link laser 50" />
  
  <node pkg="gmapping" type="slam_gmapping" name="gmapping_node" >

    <param name="maxUrange" value="4.0"/>
    <!-- Map size / start point -->
    <param name="delta" value="0.05"/>
    <param name="xmin" value="-3.0"/>
    <param name="ymin" value="-3.0"/>
    <param name="xmax" value="3.0"/>
    <param name="ymax" value="3.0"/>
    <param name="map_update_interval" value="2.0"/>
    <param name="temporalUpdate" value="2.0"/>
  </node>

  <!--used to compute frontiers by   rosrun phrontier_exploration headless phrontier_explorationpackage_extract_frontiers -->
  <node pkg="costmap_2d" type="costmap_2d_node" name="costmap_for_exploration_global">
    
    <rosparam ns="costmap" subst_value="true">
      global_frame: /map
      robot_base_frame: /base_link
      update_frequency: 0.2
      publish_frequency: 0.2
      rolling_window: false
      always_send_full_costmap: true

      # Map management parameters
      unknown_cost_value: 255
      
      robot_radius: 0.3
      plugins: 
        - {name: static, type: "costmap_2d::StaticLayer"}
        - {name: inflation,        type: "costmap_2d::InflationLayer"}  
      static:
        #Can pull data from gmapping, map_server or a non-rolling costmap            
        map_topic: /map
        subscribe_to_updates: true 
      inflation:
        inflation_radius: 0.3

    </rosparam>
  </node>
  
  <node pkg="phrontier_exploration" type="headless" name="exploration_global" args="phrontier_explorationpackage_extract_frontiers_bis">
    <param name="map_topic" value="/costmap_for_exploration_global/costmap/costmap"/>
    <param name="frontier_topic" value="phrontier_global"/>
  </node>

    <node pkg="frontier_allocation" type="frontier_allocation" name="frontier_allocation" output="screen">
      <param name="goal_tolerance" value="0.8"/>
      <param name="frontier_tolerance" value="0.2"/>
      <param name="random_frontier" value="false"/>
    </node>  
  
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find adaptive_local_planner)/rviz/robot_navigation.rviz"/>
</launch>